### PR TITLE
PR: Added support to the principal numeric Numpy types in the Variable Explorer

### LINKED ISSUE #3638: Variable explorer should display the value of numpy.int64 variables
Execute
```
import numpy
j = numpy.int64(17)
```
.

Now `j` shows up in the variable explorer, but the Value column displays `class 'numpy.int64`. It should display the value `17` instead.

(My current environment is described in #3562.)

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #3638

### CODE REVIEW

**Comment by dalthviz** [comment]
Added support in the variable explorer to the following numpy data types: `int32`, `int64`, `float32`, `float64`, `complex64`, `complex_` (shorthand for `complex128`) and `complex128`

numpy

**Comment by wtheis** [comment]
If possible, please include this in 3.0.2.
(#3638 is currently 3.1.)

File: spyder/widgets/variableexplorer/utils.py [hunk 1]
```diff
@@ -44,10 +44,12 @@ class FakeObject(object):
...
 
 #==============================================================================
-# Numpy arrays support
+# Numpy arrays and numeric support
 #==============================================================================
 try:
-    from numpy import ndarray, array, matrix, recarray
+    from numpy import ndarray, array, matrix, recarray, \
+                      int64, int32, float64, float32, complex_, \
+                      complex64, complex128
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/utils.py]: Please enclose imports in parenthesis

File: spyder/widgets/variableexplorer/utils.py [hunk 2]
```diff
@@ -292,17 +295,7 @@ def value_to_display(value, minmax=False):
...
-        elif isinstance(value, int32):
-            display = repr(value)
-        elif isinstance(value, float64):
-            display = repr(value)
-        elif isinstance(value, float32):
-            display = repr(value)
-        elif isinstance(value, complex_) or isinstance(value, complex128) :
-            display = repr(value)
-        elif isinstance(value, complex64):
+        elif True in [isinstance(value, x) for x in numeric_type_list]:
```
Comment (jitseniesen) [inline spyder/widgets/variableexplorer/utils.py]: I'd write this as `any(isinstance(value, x) for x in numeric_type_list)` or `isinstance(value, numeric_type_tuple)`; in the latter case, I think it needs to be a tuple of types, not a list of types.

Comment (ccordoba12) [inline spyder/widgets/variableexplorer/utils.py]: I also like the second option. @dalthviz, please do the corresponding change :-)

**Comment by ccordoba12** [comment]
> If possible, please include this in 3.0.2.

@wtheis, unfortunately 3.0.2 is almost done, and we don't to delay it longer.

File: spyder/widgets/variableexplorer/utils.py [hunk 3]
```diff
@@ -244,6 +247,8 @@ def unsorted_unique(lista):
 def value_to_display(value, minmax=False):
     """Convert value for display purpose"""
     try:
+        numeric_type_tuple = (int64, int32, float64, float32, \
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/utils.py]: @dalthviz, I think you don't need a continuation line (`\`) here

Comment (ccordoba12) [inline spyder/widgets/variableexplorer/utils.py]: Also, please rename this variable to `numeric_numpy_types`

File: spyder/widgets/variableexplorer/utils.py [hunk 4]
```diff
@@ -44,16 +44,19 @@ class FakeObject(object):
 
 
 #==============================================================================
-# Numpy arrays support
+# Numpy arrays and numeric support
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/utils.py]: Please change this to `Numpy arrays and numeric types support`

**Pull request merged**